modules/repartitioner.py

```diff
@@ -0,0 +1,223 @@
+            if resinRootPart.getLength(unit) == resinUpdtPart.getLength(unit):
+                #
+                # State A
+                #
+                log.debug("Running transition from State A...")
+
+                # Edit resin-updt partition
+                if not self.editPartition(targetPartition=resinUpdtPart, deltaStart=(deltasize // 2), deltaEnd=0, fstype='ext3', fslabel='resin-updt', unit=unit):
+                    log.error("increaseResinBootTo: Could not edit resin-updt partition.")
+                    return False
+
+                # Copy resin-root to resin-boot
+                log.info("increaseResinBootTo: Copying resin-root to resin-updt. This will take a while...")
+                resinRootMountPoint = getConfigurationItem(self.conf, 'General', 'host_bind_mount')
+                if not resinRootMountPoint:
+                    resinRootMountPoint = '/'
+                try:
+                    resinUpdtMountPoint = tempfile.mkdtemp(prefix='resinhup-', dir='/tmp')
+                except:
+                    log.error("increaseResinBootTo: Failed to create temporary mountpoint.")
+                    return False
+                if not mount(resinUpdtPart.path, resinUpdtMountPoint):
+                    log.error("increaseResinBootTo: Failed to mount " + resinUpdtPart.path + " to " + resinUpdtMountPoint + ".")
+                if not safeCopy(resinRootMountPoint, resinUpdtMountPoint, sync=False):
+                    log.error("increaseResinBootTo: Failed to copy files from " + resinRootMountPoint + " to " + resinUpdtMountPoint + ".")
+                    umount(resinUpdtMountPoint) # We fail anyway so don't care out return value
+                    return False
+                if not umount(resinUpdtMountPoint):
+                    log.error("increaseResinBootTo: Failed to unmount " + resinUpdtMountPoint + ".")
+                    return False
+
+                # Configure bootloader
+                if not configureBootloader(self.currentResinRootPartPath, resinUpdtPart.path, self.conf):
+                    log.error("increaseResinBootTo: Could not configure bootloader.")
+                    return False
+
+                # We reboot cause this is an intermediate repartitioning step
+                if not self.testMode:
+                    os.system('reboot')
+            else:
+                #
+                # State E
+                #
+                log.debug("Running transition from State E...")
+
+                log.error("increaseResinBootTo: Unknown filesystem state where booted from resin-boot but having different size then resin-updt.")
+                return False
+
+        elif self.currentResinRootPartPath == resinUpdtPart.path:
+            # Booted from resin-updt
+            if resinRootPart.getLength(unit) == resinUpdtPart.getLength(unit):
+                #
+                # State B
+                #
+                log.debug("Running transition from State B...")
```

Comment:
Don't you have to copy the contents of `resin-updt` to `resin-root` here before rebooting to it?